tool/cli.js

```diff
@@ -242,16 +242,24 @@ program
     })
   )
 
-  .command("preview", "Open a preview of a slug")
-  .argument("<slug>", "Slug of the document in question")
+  .command("preview", "Open a preview, given a slug or pathname or URL")
+  .argument(
+    "<slugOrPathnameOrUrl>",
+    "Slug or pathname or URL of the document in question"
+  )
   .argument("[locale]", "Locale", {
     default: DEFAULT_LOCALE,
     validator: [...VALID_LOCALES.values()],
   })
   .action(
     tryOrExit(async ({ args }) => {
-      const { slug, locale } = args;
+      const { slugOrPathnameOrUrl, locale } = args;
+      const slug = slugOrPathnameOrUrl.replace(
+        /(https:\/\/([^/]+\/en-US(\/docs)?\/)|files\/en-us\/)?(.+?)(\/index.html)?/g,
```

Comment:
I don't like this. It's hardcoded `en-US` (twice). Instead, I think you should use something like:
```
if (slugOrPathnameOrUrl.includes('://')) {
  const pathname = new URL(slugOrPathnameOrUrl).pathname.split('/')
  locale = pathname[1]
  slug = pathname.slice(2).join('/')
}
```
or something like that.

Reply:
OK will make an update for that

Reply:
I agree with @peterbe here. It would also be nice then to support relative and absolute paths.